Fakebook.Profile/Fakebook.Profile.DataAccess/ProfileRepository.cs

```diff
@@ -186,6 +190,10 @@ private async Task<DomainProfile> GetProfileInternalAsync(string email)
             }
 
             var profile = await _context.EntityProfiles
+                .Include(x => x.Following)
+                    .ThenInclude(x => x.Following)
+                .Include(x => x.Followers)
+                    .ThenInclude(x => x.User)
```

Comment:
it's a shame we're loading potentially a lot of unnecessary data here & elsewhere. keep an eye out for opportunities to reduce this sort of thing.

Reply:
@escalonn What's the best way of reducing this? My initial thought would be to have separate methods that include these lists and only call them when needed instead of every time you grab a profile. Does that seem correct?

Reply:
yeah that's a good start conceptually, but i'm not sure when they won't be needed in the current design. that judgment has to take into account the REST resource design and therefore the needs of the clients.